src/basics.md

```diff
@@ -1242,6 +1243,50 @@ fn main() {
 }
 ```
 
+[ex-datetime-arithmetic]: #ex-datetime-arithmetic
+<a name="ex-datetime-arithmetic"></a>
+## Add or subtract dates and times using checked arithmetic
+[![chrono-badge]][chrono] [![cat-date-and-time-badge]][cat-date-and-time]
+
+Calculates and displays the date and time two weeks from now using
+[`DateTime::checked_add_signed`] and the date of yesterday using
+[`DateTime::checked_sub_signed`]. The methods return None if the date and time
+cannot be calculated.
+
+Escape sequences that are available for the
+[`DateTime::format`] can be found at [`chrono::format::strftime`].
+
+```rust
+extern crate chrono;
+use chrono::{Duration, Utc};
+
+fn main() {
+    let now = Utc::now();
+    println!("The date and time are {}", now);
+
+    if let Some(two_weeks_from_now) = now.checked_add_signed(Duration::weeks(2)) {
+        println!("Two weeks from now is {}", two_weeks_from_now);
+    } else {
+        println!("Two weeks from now overflows!");
+    }
+
+    if let Some(yesterday) = now.checked_sub_signed(Duration::days(1)) {
+        println!("Yesterday was {}", yesterday.format("%a %b %e %Y"));
+    } else {
+        println!("Yesterday overflows!");
+    }
+
+    if let Some(heat_death) = now.checked_add_signed(Duration::max_value()) {
+        println!(
+            "Heat death of the universe: {}",
```

Comment:
Heat death of the universe is estimated at ~10e103 years while Duration::max_value() is 2^63 (i64::MAX) in milliseconds making it roughly 2.9243e8 years. So our statement would be a little inaccurate (94 orders of magnitude give or take ;)
Which is sad cause the idea was awesome :)

I would suggest to change it to something like ("maximal representable time" ) unless you come up with something equally awesome as heat death of the universe.

Reply:
Yeah, I tried to do it with the correct numbers at first, but realized that I'd have to delegate the heavy-lifting to a new crate, and that was a place I wasn't prepared to go to. :D I'll figure it out.